StRoot/StarGenerator/EVENT/StarGenEvent.h

```diff
@@ -205,6 +205,12 @@ class StarGenEvent : public TObject
   /// Returns the filter result
   UInt_t GetFilterResult(){ return mFilterResult; }
 
+  /// Attach a user-weight to the event
+  void AddUserWeight( double w ){ mWeights.push_back(w); }
+
+  /// Retrieve the user weights vector
+  const std::vector<double>& GetUserWeights(){ return mWeights; }
```

Comment:
```suggestion
  const std::vector<double>& GetUserWeights() const { return mWeights; }
```
If you want to be a c++ purist 🐕 `